Components: ResizableBox, add showHandle prop

ItsJonQ opened the pull request:
![Screen Capture on 2019-10-24 at 18-16-57](https://user-images.githubusercontent.com/2322354/67529829-4af57c80-f68b-11e9-9670-759c81b53996.gif)

## Description

This update adds a new prop to the ResizableBox component. This new prop `showHandle` (`bool`) determines if the draggable resize handles should be visible or not. This update provides more flexible for how `ResizableBox` can render.

## How has this been tested?
Storybook

## Screenshots

<img width="655" alt="Screen Shot 2019-10-24 at 6 18 22 PM" src="https://user-images.githubusercontent.com/2322354/67529839-55b01180-f68b-11e9-8327-1a08200460fb.png">

## Types of changes

* Add property (with default value) to `ResizableBox`
* Add  accompanying CSS className + style for that prop
* Add Storybook example + knobs 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->

Helps resolve https://github.com/WordPress/gutenberg/issues/16652

Commit:
Components: ResizableBox, Add showHandle prop

This update adds a new prop to the ResizableBox component. This new prop
`showHandle` (`bool`) determines if the draggable resize handles should be
visible or not.

Commit:
Add ResizableBox storybook example

This update adds a default story for ResizableBox. Knobs have been added
to the story to better demonstrate the effects of the props.

gziolo commented on packages/components/src/resizable-box/style.scss:
```diff
@@ -9,6 +9,10 @@
 	.components-resizable-box__container.is-selected & {
 		display: block;
 	}
+	// Show the resize handle if set in props
```
Can you combine those 2 rules which set `display:block` into one? They seem to have the same purpose.

ItsJonQ replied:
Sure thing!

gziolo commented on packages/components/src/resizable-box/stories/index.js:
```diff
@@ -0,0 +1,80 @@
+	const { content, ...restProps } = props;
+
+	return (
+		<div style={ { margin: 30 } }>
```
What do you think about refactoring Example to be `ResizableBoxWithState` to return only this part:
```js
<ResizableBox
	{ ...restProps }
	size={ {
		height,
		width,
	} }
	onResizeStop={ ( event, direction, elt, delta ) => {
		setAttributes( {
			height: parseInt( height + delta.height, 10 ),
			width: parseInt( width + delta.width, 10 ),
		} );
	} }
>
	{ children }
</ResizableBox>
```
and moving unrelated styling to the parent component?

This way we could see more code in the preview in Storybook.

gziolo commented on packages/components/src/resizable-box/index.js:
```diff
@@ -23,7 +23,8 @@ function ResizableBox( { className, ...props } ) {
 			className={ classnames(
 				'components-resizable-box__container',
-				className,
+				showHandle && 'is-show-handle',
```
Did you consider an alternative name for `is-show-handle`? It doesn't read great. `shows-handle` would be better but it probably doesn't follow the naming convention for the class names. Maybe `has-resize-handle` based on the code comment?

ItsJonQ replied:
Ah yes! `has-resize-handle` reads better. It's a struggle sometimes with these names 😂

Commit:
Merge branch 'master' into add/resizable-box-show-handle

Commit:
Rename is-show-handle to has-show-handle

Refactor styles to consolidate selected and show handle

Commit:
Refactor ResizableBox story example

Moves the inline-styled div to the parent. Allows for the Example
component to read more clearly.

ItsJonQ commented:
@gziolo Thank you for those suggestions! I just made the updates ❤️

gziolo commented on packages/components/src/resizable-box/README.md:
````diff
@@ -53,3 +53,11 @@ const Edit = ( props ) => {
 }
 ```
+
+### Props
````
By the way, this is why I like when you abstract away external code, you get the freedom to tailor it to your needs 👍

ItsJonQ replied:
Yay! Thank you 🙌

gziolo reviewed (approved):
I got super excited about testing this PR. Storybook is amazing, it makes the testing a real pleasure! 💯 

Great work on this one, thanks for addressing my feedback. This is ready to go. 

I think this PR will auto-close the linked issues in the description, so you should double-check if that is expected. Sometimes you need to be very careful because those trigger words recognized by GitHub don't look at the full context :) You wrote `helps resolve XYZ` but GitHub understood only `resolve XYZ` 😅

ItsJonQ commented:
> Storybook is amazing, it makes the testing a real pleasure!
@gziolo You are very welcome!! It's very exciting stuff. And makes it much easier for new comers like myself to work with and focus just on the UI 😊 

> You wrote helps resolve XYZ but GitHub understood only resolve XYZ

Oh gotcha! Also... TIL `resolves` is a trigger word 😮

ItsJonQ commented:
> I feel like having it comment on every build would just add a lot of noise in PRs and ping everyone unnecessarily.

Gotcha! I believe it only adds one Github comment. Still, you make a good point. If you think it'll be too noisy, we don't have to add
